radix-engine/src/blueprints/transaction_processor/package.rs

```diff
@@ -0,0 +1,75 @@
+use crate::errors::InterpreterError;
+use crate::errors::RuntimeError;
+use crate::kernel::kernel_api::KernelNodeApi;
+use crate::kernel::kernel_api::KernelSubstateApi;
+use crate::system::kernel_modules::costing::FIXED_LOW_FEE;
+use crate::types::*;
+use radix_engine_interface::api::package::TRANSACTION_PROCESSOR_RUN_IDENT;
+use radix_engine_interface::api::unsafe_api::ClientCostingReason;
+use radix_engine_interface::api::ClientApi;
+use radix_engine_interface::blueprints::transaction_processor::*;
+use radix_engine_interface::schema::BlueprintSchema;
+use radix_engine_interface::schema::FunctionSchema;
+use radix_engine_interface::schema::PackageSchema;
+
+use super::TransactionProcessorBlueprint;
+
+pub struct TransactionProcessorNativePackage;
+
+impl TransactionProcessorNativePackage {
+    pub fn schema() -> PackageSchema {
+        let mut aggregator = TypeAggregator::<ScryptoCustomTypeKind>::new();
+
+        let mut substates = Vec::new();
+        substates.push(aggregator.add_child_type_and_descendents::<()>());
```

Comment:
I didn't question this in the last PR but why do you think we should have atleast one substate per blueprint?

Reply:
It's changed in later PRs. `SELF` substates can be empty.

The original motivation of making `[0]` mandatory is for state-based access rules.

Changed auth to tolerate missing `[0]` as well.